### PR TITLE
Add initial implementation

### PR DESCRIPTION
**Pull request by rosensilva**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,81 @@
...
+        String nameSpace;
+```
+
+``Document `` contains ```org.w3c.dom.Document``` of the Request message that need to send to the SOAP backend under
+``soapAction``. This Document contains the message with the placeholders for JSON input
+
+## License
+
+```
+Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (madushadhanushka) [inline README.md]: This need to be 2022 right?

File: src/main/java/org/wso2/soaptorest/OASGenerator.java [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+                break;
+            case "long":
+            case "unsignedLong":
+                outputSchema = new IntegerSchema();
+                outputSchema.setType("integer");
+                outputSchema.setFormat("int64");
+                break;
+            case "boolean":
+                outputSchema = new BooleanSchema();
+                outputSchema.setType("boolean");
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/soaptorest/OASGenerator.java:382]: `break` statement is missing?

File: src/main/java/org/wso2/soaptorest/SOAPRequestBodyGenerator.java [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+        boolean isNamespaceQualified = false;
+        boolean isRootComplexType = false;
+        Document doc;
+        try {
+            docBuilder = docFactory.newDocumentBuilder();
+            doc = docBuilder.newDocument();
+            Element rootElement = null;
+            if (SOAPToRESTConstants.SOAP_RPC_MESSAGE_TYPE.equalsIgnoreCase(soapMessageType) ||
+                    SOAPToRESTConstants.SOAP_RPC_MESSAGE_TYPE.equalsIgnoreCase(soapStyle) ||
+                    parameterJsonPathMapping.size() == 0) {
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/soaptorest/SOAPRequestBodyGenerator.java]:
```suggestion
                    parameterJsonPathMapping.isEmpty()) {
```

File: src/main/java/org/wso2/soaptorest/SOAPRequestBodyGenerator.java [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+                            } else {
+                                doc.appendChild(element);
+                            }
+                            prevElement = element;
+                        }
+                        elemPos++;
+                    }
+                }
+            }
+            if (parameterJsonPathMapping.size() == 0) {
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/soaptorest/SOAPRequestBodyGenerator.java]:
```suggestion
            if (parameterJsonPathMapping.isEmpty()) {
```

File: src/main/java/org/wso2/soaptorest/OASGenerator.java [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+import io.swagger.v3.oas.models.*;
+import io.swagger.v3.oas.models.info.Info;
+import io.swagger.v3.oas.models.media.*;
+import io.swagger.v3.oas.models.parameters.RequestBody;
+import io.swagger.v3.oas.models.responses.ApiResponse;
+import io.swagger.v3.oas.models.responses.ApiResponses;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.soaptorest.models.*;
```
Comment (madushadhanushka) [inline src/main/java/org/wso2/soaptorest/OASGenerator.java:29]: Shall we remove the usage of *

File: README.md (hunk 1 above)
Comment (rosensilva) [reply README.md]: fixed

File: src/main/java/org/wso2/soaptorest/OASGenerator.java (hunk 5 above)
Comment (rosensilva) [reply src/main/java/org/wso2/soaptorest/OASGenerator.java:29]: all of the classes in the package are being used by the class. Intellij automatically does this

**Comment by madushadhanushka** [comment]
Please remove following files
src/test/.DS_Store
src/test/resources/.DS_Store

**Review by madushadhanushka — APPROVED** [review]

**Review by SanojPunchihewa — APPROVED** [review]

**Pull request merged**